torchgeo/datasets/l8biome.py

```diff
@@ -0,0 +1,265 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""L8 Biome dataset."""
+
+import glob
+import os
+from typing import Any, Callable, Dict, List, Optional, cast, Sequence
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.colors import ListedColormap
+from PIL import Image
+from rasterio.crs import CRS
+from torch import Tensor
+from torch.utils.data import Dataset
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class L8Biome(RasterDataset):
+    """L8 Biome datasets.
+
+    The `L8 Biome <https://landsat.usgs.gov/landsat-8-cloud-cover-assessment-validation-data>`__ dataset # noqa: E501
+    is a cloud validation dataset of Pre-Collection Landsat 8 
+    Operational Land Imager (OLI) Thermal Infrared Sensor (TIRS) 
+    terrain-corrected (Level-1T) scenes.
+
+    Dataset features:
+
+    * images evenly divided between eight unique biomes
+    * 5 cloud cover categories
+
+    Dataset format:
+
+    * Each cloud mask is in ENVI binary format.
+    Includes all bands from the original Landsat Level-1 data product (GeoTIFF),
+    and its associated Level-1 metadata (MTL.txt file).
+    * Interpretation for bits in each manual mask are as follows:
+    0: Fill, 64: Cloud Shadow, 128: Clear, 192: Thin Cloud, 255: Cloud
+
+    If you use this dataset in your research, please cite the following:
+
+    * https://doi.org/10.5066/F7251GDH
+    * https://doi.org/10.1016/j.rse.2017.03.026
+
+    .. versionadded:: 0.5
+    """
+
+    url = "https://huggingface.co/datasets/torchgeo/l8biome/blob/main/{}.tar.gz"
+
+    filenames_to_md5 = {
+        "barren": "bb446fda3f6af50930849bb135e99f9c",
+        "forest": "21505d878abac830890ea84abddc3c46",
+        "grass_crops": "33d0c553357f5a439aa85a45916ac89a",
+        "shrubland": "f19afc6dfa818ee3868e7040441d4c6d",
+        "snow_ice": "d7b56084e6267ee114419efdc7f664c9",
+        "urban": "b5f6aabbb380e108c408a8ea5dae3835",
+        "water": "d143049ef64e6e681cea380dd84680e9",
+        "wetlands": "bff0d51db84e26a2a8e776c83ab2d331",
+    }
+
+    filename_glob = "LC*_B2.TIF"
+
+    separate_files = True
+    rgb_bands = ["B2", "B3", "B4"]
```

Comment:
These are BGR, not RGB. See https://developers.google.com/earth-engine/datasets/catalog/LANDSAT_LC08_C02_T1_TOA